### ISSUE TITLE
Issues compiling neuron with newest dependencies

### ISSUE THREAD

**Issue by maralorn** [srid/neuron#475]
I encounter the following problem when trying to compile neuron with a very fresh nixpkgs.
I primarily drop this here to track this issue. My assumption is, that this is a problem with missing instances from some dependencies. I have seen similar issues in the test-suite of `aeson-gadt-th`.

Right now I don‘t know, what this `Unit` type is, for which apparently the instance is missing. Will update here once I investigated.

```
src/lib/Neuron/Zettelkasten/Zettel.hs:127:1: error:
    • Could not deduce (FromJSON (Unit [TagPattern]))
        arising from a use of ‘parseJSON’
      from the context: FromJSON TagPattern
        bound by the instance declaration
        at src/lib/Neuron/Zettelkasten/Zettel.hs:127:1-28
    • In a stmt of a 'do' block: Unitx_a2J02 <- parseJSON _v'_a2J0c
      In the expression:
        do Unitx_a2J02 <- parseJSON _v'_a2J0c
           return (Some (ZettelQuery_Tags x_a2J02))
      In a case alternative:
          "ZettelQuery_Tags"
            -> do Unitx_a2J02 <- parseJSON _v'_a2J0c
                  return (Some (ZettelQuery_Tags x_a2J02))
    |
127 | deriveJSONGADT ''ZettelQuery
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/lib/Neuron/Zettelkasten/Zettel.hs:127:1: error:
    • Could not deduce (ToJSON (Unit Value))
        arising from a use of ‘toJSON’
      from the context: ToJSON TagPattern
        bound by the instance declaration
        at src/lib/Neuron/Zettelkasten/Zettel.hs:127:1-28
      or from: r ~ Map Tag Natural
        bound by a pattern with constructor:
                   ZettelQuery_Tags :: [TagPattern] -> ZettelQuery (Map Tag Natural),
                 in a case alternative
        at src/lib/Neuron/Zettelkasten/Zettel.hs:127:1-28
    • In the expression:
        toJSON ("ZettelQuery_Tags" :: String, Unit toJSON x_a2IYG)
      In a case alternative:
          ZettelQuery_Tags x_a2IYG
            -> toJSON ("ZettelQuery_Tags" :: String, Unit toJSON x_a2IYG)
      In the expression:
        \case
          ZettelQuery_ZettelByID x_a2IY4 x_a2IY5
            -> toJSON
                 ("ZettelQuery_ZettelByID" :: String,
                (toJSON x_a2IY4, toJSON x_a2IY5))
          ZettelQuery_ZettelsByTag x_a2IYn x_a2IYo x_a2IYp
            -> toJSON
                 ("ZettelQuery_ZettelsByTag" :: String,
                (toJSON x_a2IYn, toJSON x_a2IYo, toJSON x_a2IYp))
          ZettelQuery_Tags x_a2IYG
            -> toJSON ("ZettelQuery_Tags" :: String, Unit toJSON x_a2IYG)
          ZettelQuery_TagZettel x_a2IZ3
            -> toJSON ("ZettelQuery_TagZettel" :: String, Unit toJSON x_a2IZ3)
    |
127 | deriveJSONGADT ''ZettelQuery
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by ali-abrar** [comment]
You should be able to fix this by upgrading to [v0.2.5.0 of aeson-gadt-th](https://github.com/obsidiansystems/aeson-gadt-th/releases/tag/v0.2.5.0)

**Comment by maralorn** [comment]
Thanks, that fixed it!
